File: src/main/java/com/graqr/threshr/model/redsky/product/Product.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record Product(
        @JsonProperty("__typename") String typename,
        String tcin,
        @Nullable @JsonProperty("original_tcin") String originalTcin,
        Category category,
        @JsonProperty("ratings_and_reviews") RatingsAndReviews ratingsAndReviews,
        Item item,
        @JsonProperty("finds_stories") List<FindsStory> findsStories,
        @Nullable @JsonProperty("finds_posts") List<FindsPost> findsPosts,
        Price price,
        List<?> promotions) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/Product.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Product(
    @param:JsonProperty("__typename") val typename: String,
    val tcin: String,
    @param:Nullable @param:JsonProperty("original_tcin") val originalTcin: String,
    val category: Category,
    @param:JsonProperty("ratings_and_reviews") val ratingsAndReviews: RatingsAndReviews,
    val item: Item,
    @param:JsonProperty("finds_stories") val findStories: List<FindsStory>,
    @param:Nullable @param:JsonProperty("finds_posts") val findsPosts: List<FindsPost>,
    val price: Price,
    val promotions: List<Any>
) {

}